pkg/apis/registry/types.go

```diff
@@ -40,6 +40,8 @@ type RegistryCache struct {
 	SecretReferenceName *string
 	// Proxy contains settings for a proxy used in the registry cache.
 	Proxy *Proxy
+	// HighAvailability defines if the StatefulSet is scaled with the [High Availability](https://gardener.cloud/docs/gardener/high-availability/#system-components) feature.
+	HighAvailability *bool
```

Comment:
I would suggest the new field structure to be:
```yaml
highAvailability:
  enabled: true
```

This makes the API open for future backwards-compatible modifications - for example add another high availability related field.